File: Sorting_Algorithms.py
# 1. Insertion Sort
def insertion_sort(arr):
    for i in range(1, len(arr)):
        temp = arr[i]
        j = i
        while j>0 and arr[j-1]>temp:
            arr[j] = arr[j-1]
            j = j-1
        arr[j] = temp
    return arr

# Runner Code
arr = list(map(int, input().split(" ")))
print("sorted array is", insertion_sort(arr))

File: Sorting_Algorithms.cpp
#include<iostream>
using namespace std;
int insertion_sort(int len, int arr[])
{
    for(int i = 1;i<len;i++)
    {
        int temp = arr[i];
        int j = i;
        while(j>0 && arr[j-1]>temp)
        {
            arr[j] = arr[j-1];
            j = j-1;
        }
        arr[j] = temp;
    }
    for(int i = 0;i<len;i++)
    {
        cout<<arr[i];
    }
}

int main()
{
    int len;
    cin>>len;
    int arr[len];
    for(int i = 0;i<len;i++)
    {
        cin>>arr[i];
    }
    insertion_sort(len, arr);
}
